cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java

```diff
@@ -75,11 +76,9 @@ public void getAuthURL(HttpServiceRequest request, HttpServiceResponder responde
                          @QueryParam("redirect_url") String redirectURL) {
     try {
       OAuthProvider oauthProvider = getProvider(provider);
-      String response = String.format(
-          "%s?response_type=code&client_id=%s&redirect_uri=%s&scope=refresh_token%%20api",
-          oauthProvider.getLoginURL(),
-          oauthProvider.getClientCredentials().getClientId(),
-          redirectURL);
+      String response = oauthProvider.getLoginURL()
+          .replace(CLIENT_ID, oauthProvider.getClientCredentials().getClientId())
```

Comment:
Please don’t use string replace like this. The necessary information can be passed to the provider to form the url and return

Reply:
Fixed